prusti-specs/src/specifications/preparser.rs

```diff
@@ -289,557 +58,547 @@ impl Parse for SpecEntArgs {
-                    typ: var.typ,
-                    name: var.name
+                if let Some(stream) = self.consume_group(Delimiter::Bracket) {
+                    Parser::from_token_stream_and_span(stream, self.last_span).extract_entailment_rhs(lhs, vars)
+                } else {
+                    Err(self.error_expected("`[`"))
+                }
+            } else {
+                Ok(AssertionWithoutId {
+                    kind: box common::AssertionKind::Expr(lhs)
                 })
             }
-
-            // parse body
-            let token_stream = stream.create_stream_until(",");
-            let mut parser = Parser::from_token_stream(token_stream);
-            let body = parser.extract_assertion()?;
-
-            // create triggers in case they are not present
-            let mut trigger_set = TriggerSet(vec![]);
-
-            // parse triggers (check if they are present at all)
-            if stream.peek_operator(",") {
-                stream.check_and_consume_operator(",");
-                if !stream.check_and_consume_keyword("triggers") {
-                    return Err(self.error_expected_triggers());
-                }
-                if !stream.check_and_consume_operator("=") {
-                    return Err(self.error_expected_equals());
-                }
-                let token_stream = stream.create_stream();
-
-                let maybe_arr: Result<syn::ExprArray, Error> = syn::parse2(token_stream);
-                if let Err(err) = maybe_arr {
-                    self.input.span = err.span();
-                    return Err(self.error_expected_tuple());
-                }
-                let arr = maybe_arr.unwrap();
-                self.input.span = arr.span();
-
-                let mut vec_of_triggers = vec![];
-                for item in arr.elems {
-                    if let syn::Expr::Tuple(tuple) = item {
-                        vec_of_triggers.push(
-                            Trigger(tuple.elems
-                                .into_iter()
-                                .map(|x| ExpressionWithoutId {
-                                    id: (),
-                                    spec_id: common::SpecificationId::dummy(),
-                                    expr: x })
-                                .collect()
-                            )
-                        );
-                    }
-                    else {
-                        self.input.span = item.span();
-                        return Err(self.error_expected_tuple());
+        }
+    }
+    fn extract_entailment_rhs(&mut self, lhs: ExpressionWithoutId, vars: Vec<Arg>) -> syn::Result<AssertionWithoutId> {
+        let mut pres = vec![];
+        let mut posts = vec![];
+        let mut first = true;
+        while !self.tokens.is_empty() {
+            if first || self.consume_operator(",") {
+                first = false;
+                    if self.consume_keyword("requires") {
```

Comment:
Wrong indentation.

Reply:
That's an artifact of how Github handles long lines - I'll add a line break.

Reply:
Addressed by c77b6d0b1638c36b81eddb1e87af37c3f1fff212